hummingbot/market/bitfinex/bitfinex_market.pyx

```diff
@@ -0,0 +1,1392 @@
+
+                    for wallet in wallets:
+                        wallet_type = wallet[0]
+                        if (wallet_type != "exchange"):
+                            continue
+
+                        asset_name = wallet[1]
+                        balance = wallet[2]
+                        balance_available = wallet[4]
+
+                        self._account_balances[asset_name] = Decimal(balance or 0)
+                        self._account_available_balances[asset_name] = Decimal(balance_available or 0)
+                        # remote_asset_names.add(asset_name)
+
+                    # asset_names_to_remove = local_asset_names.difference(remote_asset_names)
+                    # for asset_name in asset_names_to_remove:
+                    #     del self._account_available_balances[asset_name]
+                    #     del self._account_balances[asset_name]
+
+                # else (previous author)
+                else:
+                    content = self.parse_message_content(*event_message.content)
+                    if not content:
+                        continue
+                    event_type = content.get("type")
+                    # str - because from exchange come int; order.exchange_order_id is str
+                    exchange_order_ids = [
+                        str(content.get("order_id")),
+                        str(content.get("maker_order_id")),
+                        str(content.get("taker_order_id"))
+                    ]
+
+                    tracked_order = None
+                    for order in self._in_flight_orders.values():
+                        if order.exchange_order_id in exchange_order_ids:
+                            tracked_order = order
+                            break
+                    if tracked_order is None:
+                        continue
+
+                    order_type_description = tracked_order.order_type_description
+                    execute_price = Decimal(content.get("price", 0.0))
+                    execute_amount_diff = s_decimal_0
+
+                    # trade update is like rollup state. each event increment
+                    # amount and price. When amount is 0, it will meant order fill.
+                    if event_type in [ContentEventType.TRADE_UPDATE]:
+                        # amount_come - negative is sell, positive - buy.
+                        # for checking that order is fill
+                        amount_come = Decimal(content["amount"]).quantize(Decimal('1e-8'))
+                        execute_amount_diff = (abs(tracked_order.amount) - abs(amount_come)).quantize(Decimal('1e-8'))
+                        tracked_order.executed_amount_base += abs(amount_come)
+                        tracked_order.executed_amount_quote += abs(amount_come) * execute_price
+
+                    if execute_amount_diff == s_decimal_0.quantize(Decimal('1e-8')) \
```

Comment:
@nionis 
That line causes the bot to keep executing `_update_order_status` abnormally.
I believe the condition there ought to check if `execute_amount_diff > s_decimal_0`.